app/controllers/api/v1/agent_auth_base_controller.rb

```diff
@@ -5,12 +5,27 @@ class Api::V1::AgentAuthBaseController < Api::V1::BaseController
   include DeviseTokenAuth::Concerns::SetUserByToken
 
   skip_before_action :verify_authenticity_token
-  before_action :authenticate_api_v1_agent_with_token_auth!
+  before_action :authenticate_api_v1_agent_with_token_auth!, unless: :shared_secret_is_valid?
+  before_action :authenticate_agent_with_shared_secret, if: :shared_secret_is_valid?
 
   def pundit_user
     AgentOrganisationContext.new(current_agent, current_organisation)
   end
 
+  def shared_secret_is_valid?
+    return false if request.headers["shared-secret-for-agents-auth"].nil?
```

Comment:
J'ai du mal à savoir ce qu'il se passe lorsque l'on passe ce header avec une valeur vide. Je suggère que le critère soir la la présence du header, même vide, c'est à dire que si on passe le header vide, c'est comme passer un mauvais mot de passe.

Le critère deviendrait alors :

```suggestion
    return false if request.headers.include?("shared-secret-for-agents-auth")
```